sdks/java/core/src/test/java/org/apache/beam/sdk/testing/TestPipelineTest.java

```diff
@@ -52,280 +59,496 @@
-  }
 
-  private PCollection<String> addTransform(final PCollection<String> pCollection) {
-    return pCollection.apply(
-        MapElements.via(
-            new SimpleFunction<String, String>() {
+      private static final transient ExpectedException exception = ExpectedException.none();
 
-              @Override
-              public String apply(final String input) {
-                return DUMMY;
-              }
-            }));
-  }
+      static {
+        System.setProperty(TestPipeline.PROPERTY_USE_DEFAULT_DUMMY_RUNNER, Boolean.TRUE.toString());
+        exception.expect(IllegalStateException.class);
+      }
 
-  @Test
-  public void testPipelineRunMissing() throws Throwable {
-    exception.expect(TestPipeline.PipelineRunMissingException.class);
-    PAssert.that(pCollection()).containsInAnyOrder(DUMMY);
-    // missing pipeline#run
-  }
+      private final transient TestPipeline pipeline = TestPipeline.create();
 
-  @Test
-  public void testPipelineHasAbandonedPAssertNode() throws Throwable {
-    exception.expect(TestPipeline.AbandonedNodeException.class);
-    exception.expectMessage("PAssert");
+      @Rule
+      public final transient RuleChain chain = RuleChain.outerRule(exception).around(pipeline);
 
-    final PCollection<String> pCollection = pCollection();
-    PAssert.that(pCollection).containsInAnyOrder(DUMMY);
-    pipeline.run().waitUntilFinish();
+      @Category(NeedsRunner.class)
+      @Test
+      public void testConflictNeedsRunnerWithCrashingRunner() throws Exception {
 
-    // dangling PAssert
-    PAssert.that(pCollection).containsInAnyOrder(DUMMY);
-  }
+        addTransform(pCollection(pipeline));
 
-  @Test
-  public void testPipelineHasAbandonedPTransformNode() throws Throwable {
-    exception.expect(TestPipeline.AbandonedNodeException.class);
-    exception.expectMessage("PTransform");
+        // @NeedsRunner annotation with a CrashingRunner indicates a configuration issue
+      }
 
-    final PCollection<String> pCollection = pCollection();
-    PAssert.that(pCollection).containsInAnyOrder(DUMMY);
-    pipeline.run().waitUntilFinish();
+      @Category(RunnableOnService.class)
+      @Test
+      public void testConflictRunnableOnServiceWithCrashingRunner() throws Exception {
 
-    // dangling PTransform
-    addTransform(pCollection);
-  }
+        addTransform(pCollection(pipeline));
 
-  @Test
-  public void testNormalFlowWithPAssert() throws Throwable {
-    PAssert.that(pCollection()).containsInAnyOrder(DUMMY);
-    pipeline.run().waitUntilFinish();
-  }
+        // @RunnableOnService annotation with a CrashingRunner indicates a configuration issue
+      }
 
-  @Test
-  public void testAutoAddMissingRunFlow() throws Throwable {
-    PAssert.that(pCollection()).containsInAnyOrder(DUMMY);
-    // missing pipeline#run, but have it auto-added.
-    pipeline.enableAutoRunIfMissing(true);
-  }
+    }
+
+
+    private static List<Object[]> extractTests(final Class<?> testClass) {
```

Comment:
The tests should be run in the context of the `runners/direct` module, at minimum, so the same signal should be obtained. Failing that, adding an additional category and explicitly including it during the execution of the core SDK should also work (even if it adds an additional Surefire execution to the core SDK)